exercises/rational-numbers/example.py

```diff
@@ -1,7 +1,5 @@
 from __future__ import division
 
-from fractions import gcd
```

Comment:
I would recommend not abandoning builtin implementations, since they are known to be efficient (and are occasionally implemented in C, providing better performance potential than any user-defined implementation can provide).

The deprecation warning should able to be resolved with the following snippet:

```Python
try:
	from math import gcd
except ImportError:
	from fractions import gcd
```

Reply:
That seems like a better solution, thanks!
In the current `example.py`,  when math.gcd is used there is a failure in one of the tests - `test_divide_two_negative` (not sure why). My solution passes the tests with `math.gcd`. Rather than trying to modify the current solution, could I suggest the [other solution](https://exercism.io/tracks/python/exercises/rational-numbers/solutions/cb3071d7dbe84854a835639ca996a153) instead?

Reply:
Am I missing something? Both look the same:

```Python
# Current
def __truediv__(self, other):
     return Rational(self.numer * other.denom, self.denom * other.numer)

# Other
def __truediv__(self, other):
    return Rational(self.numer * other.denom, self.denom * other.numer)

Reply:
Yes. Looking into it, I think the problem is with the way the current solution handles negative numbers in "_reduce". The `if` only modifies the case where n is positive and d is negative. The case where both are negative is not treated correctly.

Reply:
Ah I figured it out. gcd(-2,-4) works differently. In fractions it returns -2 in math it returns (the expected) 2. Perhaps in the case of `gcd` here, it would be better to go with the user defined implementation, even if it would be slightly slower? It may get confusing to check the different versions when the two gcd functions behave differently

Reply:
> It may get confusing to check the different versions when the two gcd functions behave differently

Or, for safety, just wrap the `gcd` call in `math.abs` (and include a comment saying why).

Reply:
Even better, I will try and do that later. Thanks!